Horizon/Horizon/Sources/Features/Learn/View/CourseDetailsViewModel.swift

```diff
@@ -29,27 +29,30 @@ final class CourseDetailsViewModel: ObservableObject {
 
     // MARK: - Private
 
-    private let environment: AppEnvironment
+    private let router: Router
+    private let onTabBarVisibility: (Bool) -> Void
     private var subscriptions = Set<AnyCancellable>()
 
     // MARK: - Init
 
     init(
-        environment: AppEnvironment,
-        course: HCourse
+        router: Router,
+        course: HCourse,
+        onTabBarVisibility: @escaping (Bool) -> Void
```

Comment:
Are we ever going to call this with a false value? If not we could simply call it `showTabBar`.